packages/frontend/src/components/DataViz/config/CartesianChartSeries.tsx

```diff
@@ -46,217 +41,227 @@ export const CartesianChartSeries = ({
         selectCurrentCartesianChartState(state, selectedChartType),
     );
 
-    const series: ConfigurableSeries[] = useMemo(() => {
-        if (!currentConfig?.fieldConfig?.y) {
-            return [];
+    const series = useAppSelector((state) =>
+        selectCurrentCartesianChartState(state, selectedChartType),
+    )?.series;
+
+    const groupedSeries: Record<string, ConfigurableSeries[]> = useMemo(() => {
+        if (!series) {
+            return {};
         }
-        return currentConfig?.fieldConfig?.y.map((f, index) => {
-            const foundSeries = Object.values(
-                currentConfig?.display?.series || {},
-            ).find((s) => s.yAxisIndex === index);
 
-            const seriesFormat = foundSeries?.format;
-            const seriesLabel = foundSeries?.label;
-            const seriesColor = foundSeries?.color;
-            const seriesType = foundSeries?.type;
-            const seriesValueLabelPosition = foundSeries?.valueLabelPosition;
-            const seriesWhichYAxis = foundSeries?.whichYAxis;
-            return {
-                reference: f.reference,
-                format: seriesFormat,
-                label:
-                    seriesLabel ??
-                    friendlyName(`${f.reference}_${f.aggregation}`),
-                color: seriesColor ?? colors[index],
-                type: seriesType,
-                valueLabelPosition: seriesValueLabelPosition,
-                whichYAxis: seriesWhichYAxis,
-            };
-        });
-    }, [colors, currentConfig?.display?.series, currentConfig?.fieldConfig?.y]);
+        return series.reduce<Record<string, ConfigurableSeries[]>>(
+            (acc, s, index) => {
+                const foundSeries =
+                    currentConfig?.display?.series?.[s.pivotColumnName];
+
+                const seriesFormat = foundSeries?.format;
+                const seriesLabel = foundSeries?.label;
+                const seriesColor = foundSeries?.color;
+                const seriesType = foundSeries?.type;
+                const seriesValueLabelPosition =
+                    foundSeries?.valueLabelPosition;
+                const seriesWhichYAxis = foundSeries?.whichYAxis;
+
+                const config = {
+                    reference: s.pivotColumnName,
+                    format: seriesFormat,
+                    label: seriesLabel ?? friendlyName(s.pivotColumnName),
+                    color: seriesColor ?? colors[index],
+                    type: seriesType,
+                    valueLabelPosition: seriesValueLabelPosition,
+                    whichYAxis: seriesWhichYAxis,
+                };
+
+                if (!acc[s.pivotColumnName]) {
+                    acc[s.pivotColumnName] = [];
+                }
+
+                acc[s.pivotColumnName].push(config);
+
+                return acc;
```

Comment:
monir, can be immutable:
```suggestion
                return {
                    ...acc,
                    [s.pivotColumnName]: [...(acc[s.pivotColumnName] || []), config],
                };
```

Reply:
Done